src/lib/storage/proposal/devicegraph_generator.rb

```diff
@@ -92,6 +94,35 @@ def provide_space(volumes, initial_graph, disk_analyzer)
           result_graph
         end
 
+        # Copy of the volumes list with some extra information inferred from the
+        # space maker.
+        #
+        # It enforces reuse of UUIDs and labels from the deleted swap
+        # partitions.
+        #
+        # @param volumes [PlannedVolumesList] original list of volumes
+        # @param space_maker [SpaceMaker] an instance in which
+        #       SpaceMaker#provide_space has already been called
+        # @return [PlannedVolumesList]
+        def refine_volumes(volumes, space_maker)
+          refined = volumes.deep_dup
+
+          deleted_swaps = space_maker.deleted_partitions.select do |part|
+            part.id == ::Storage::ID_SWAP
+          end
+          new_swap_volumes = refined.select { |vol| !vol.reuse && vol.mount_point == "swap" }
+
+          new_swap_volumes.each_with_index do |swap_volume, idx|
+            deleted_swap = deleted_swaps[idx]
+            break unless deleted_swap
+
+            swap_volume.uuid = deleted_swap.filesystem.uuid
+            swap_volume.label = deleted_swap.filesystem.label
+          end
+
+          refined
+        end
+
```

Comment:
So, you basically keep a collection of old names around and pick from that pool if you need some. Just one question: can it happen that a label you resurrect is already in use?